### PR TITLE
Making Papa require friendly.

### PR DESCRIPTION
**Pull request by mnibecker**
Added functionality to return Papa obj if app is using require.

### CODE REVIEW

**Comment by mnibecker** [comment]
Additionally I was thinking about add an else to return set `global.Papa = Papa` and not have Papa automatically inserted into the global namespace (which I think is more correct), but I'm concerned about this breaking some require apps that just import using `window.Papa`

File: papaparse.js [hunk 1]
```diff
@@ -1432,4 +1432,11 @@
 	{
 		return typeof func === 'function';
 	}
+
+	// if requirejs is available, return Papa
+	if (typeof define === "function" && define.amd) {
```
Comment (mholt) [inline papaparse.js]: For consistency, let's use isFunction (defined just above your edit, actually). And if `define` is defined on the `window` object (accessible as `global` inside the Papa closure), let's explicitly say `global.define` to make it clear.

**Comment by mholt** [comment]
Thanks for the PR -- I've never used RequireJS so I'll trust you to make the integration that's best.

Sorry it took me a while to get around to this.

As for setting `global.Papa = Papa`, I don't think that's necessary; `global` is `window` (at least, in the browser environment).

File: papaparse.js [hunk 2]
```diff
@@ -1432,4 +1432,11 @@
...
+	    define('papaparse', [], function() {
```
Comment (fearphage) [inline papaparse.js]: Is there a reason to name it? Why can't it be an anonymous define?

```` javascript
define(function() {
  return global.Papa;
});
````

File: papaparse.js [hunk 3]
```diff
@@ -1432,4 +1432,11 @@
...
+	    	return global.Papa;
+    	});
```
Comment (fearphage) [inline papaparse.js]: Indentation is off.

File: papaparse.js (hunk 2 above)
Comment (mnibecker) [inline papaparse.js]: good call, I'll leave it anonymous

**Comment by mnibecker** [comment]
Sorry it took me so long to get this. Made the recommended changes

**Comment by mholt** [comment]
Thanks!

**Pull request merged**